__tests__/tests/components/CommentInput.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {Platform} from 'react-native';
+
+import {CommentInput} from 'components';
+
+describe('<CommentInput />', () => {
+
+  let defaultProps = {
+    users: [],
+    userHasPushPermission: true,
+    issueLocked: false,
+    locale: '',
+    onSubmit: () => {},
+  };
+
+  it("should render 'TextInput' and TouchableOpacity if user can post", () => {
+
+    const wrapper = shallow(
+      <CommentInput {...defaultProps}/>
+    );
+
+    expect(wrapper.find('TextInput').length).toEqual(1);
+    expect(wrapper.find('TouchableOpacity').length).toEqual(1);
+    expect(wrapper.find('Text').length).toEqual(0);
+  });
+
+  it("should not render 'TextInput' and TouchableOpacity if user can post", () => {
+
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={false}
+        issueLocked={true}/>
+    );
+
+    expect(wrapper.find('TextInput').length).toEqual(0);
+    expect(wrapper.find('TouchableOpacity').length).toEqual(0);
+    expect(wrapper.find('Text').length).toEqual(1);
+  });
+
+  it('should update the state text if value change', () => {
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={true}
+        issueLocked={false}/>
```

Comment:
Are these 2 props the same with `defaultProps`?

Reply:
Yes, there are. I'll use `defaultProps`.